projects/counters/src/main.c

```diff
@@ -0,0 +1,39 @@
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+static void prv_counter_callback(SoftTimerId soft_timer_id, void *context) {
+  Counters *counter_ptr = (Counters *)context;
+  uint8_t *counter_a_ptr = &(counter_ptr->counter_a);
+  uint8_t *counter_b_ptr = &(counter_ptr->counter_b);
+
+  if ((*counter_a_ptr == 0) || ((*counter_a_ptr) % 2 == 1)) {
```

Comment:
Since counter A is increment either way, we can simply take it out of the if else statement

Reply:
We can then just increment counter B if counter A is divisible by 2